test/common/markdown/markdown.tests.ps1

```diff
@@ -7,101 +7,112 @@ Write-Verbose -message "PSScriptRoot: $PSScriptRoot" -Verbose
-                                        "be sure that you have installed nodejs and have " + `
-                                        "run 'npm install -g gulp' in order to have this " + `
-                                        "text execute.")
-            }
-            finally
+                $NpmInstalled = "Installed"
+                Write-Verbose -Message "NPM is checking Gulp is installed. This may take a few moments." -Verbose
+                $null = Start-Process `
+                    -FilePath "npm" `
+                    -ArgumentList @('install','--silent') `
+                    -Wait `
+                    -WorkingDirectory $PSScriptRoot `
+                    -PassThru `
+                    -NoNewWindow
+                $null = Start-Process `
+                    -FilePath "npm" `
+                    -ArgumentList @('install','-g','gulp','--silent') `
+                    -Wait `
+                    -WorkingDirectory $PSScriptRoot `
+                    -PassThru `
+                    -NoNewWindow
+            } 
+            elseif( -not $env:AppVeyor)
             {
-                Pop-Location
+                Write-Warning "Node and npm are required to run this test"
+                $skip = $true
             }
+        }
+    }
 
-            $LASTEXITCODE | Should beexactly 0
+    AfterAll {
+        if ( !$skip ) 
+        {
+            # We're using this tool to delete the node_modules folder because it gets too long
+            # for PowerShell to remove.
+            $null = Start-Process `
+                -FilePath "npm" `
+                -ArgumentList @('install','rimraf','-g','--silent') `
+                -Wait `
+                -WorkingDirectory $PSScriptRoot `
+                -PassThru `
+                -NoNewWindow
+            $null = Start-Process `
+                -FilePath "rimraf" `
+                -ArgumentList @(Join-Path -Path $PSScriptRoot -ChildPath 'node_modules') `
+                -Wait `
+                -WorkingDirectory $PSScriptRoot `
+                -PassThru `
+                -NoNewWindow
+        }
+    }
 
-            $mdIssuesPath = Join-Path -Path $PSScriptRoot -ChildPath "markdownissues.txt"
+    It "Should not have errors in any markdown files" -Skip:$skip {
+        $NpmInstalled | should BeExactly "Installed"
+        $mdErrors = 0
+        Push-Location -Path $PSScriptRoot
+        try
+        {
+            $docsToTest = @(
+                './*.md'
+                './docs/installation/*.md'
+            )
+            $filter = ($docsToTest -join ',')
+            Write-Verbose "Filter: $filter" -Verbose
+            &"gulp" test-mdsyntax --silent `
+                --rootpath $repoRootPath `
+                --filter $filter
 
-            Write-Verbose "$mdIssuesPath should exist" -Verbose
-            $mdIssuesPath | should exist
+            Start-Sleep -Seconds 3
```

Comment:
These tests have been in the DSCResources repos for several weeks with this delay without issue.  I assume the only issues is the time it takes for the file to flush to disk.  A few second should be sufficient a vast majority of the time.